src/entities/priceImpact/index.ts

```diff
@@ -200,6 +218,142 @@ export class PriceImpact {
         }
     };
 
+    /**
+     * Alternative for calculating price impact on add liquidity unbalanced operations
+     *
+     * Note: works on balancer v2 only because it relies on Remove Liquidity Unbalanced
+     * @param input same input used in the corresponding add liquidity operation
+     * @param poolState same pool state used in the corresponding add liquidity operation
+     * @returns price impact amount
+     */
+    static addLiquidityUnbalancedAlternative = async (
```

Comment:
When would someone choose addLiquidityUnbalancedAlternative over addLiquidityUnbalanced? Do we need to keep both?

Reply:
I added the reason to the PR description:
> I thought it would be interesting to have it in the repo because it serves the purpose of validating the more generic price impact calculation for Add Liquidity Unbalanced, which is the only one that uses a modified ABA approach.

This alternative method actually follows more precisely the principle of the ABA method, but is only available for v2.
The main one is an adaptation that works with both v2 and v3 because it doesn't rely on Remove Liquidity Unbalanced.

I can see that it may cause confusion, and that's why I added comments explaining where they can be applied.
Let me know if you still think we're better off without it and I'll remove it over here.

Reply:
I wasn't really sure who would find it useful for validation? - If its just us internally then I think we should move it out to a test helper (assuming validation is during testing?). If we're not using it for testing then I'd remove it completely, could always copy the snippet to Notion if you think it'll be useful in future.

For the consumer I definitely think it'll be confusing which to use - if we have something that can cover V2 & V3 we should just use that by default and take the decision away 👍.

Reply:
I agree that it's confusing for the consumer and that it's beneficial to take the decision away 👍 
I like the idea of having within a test. I'll do it over here.